mattermost-plugin/server/notifications.go

```diff
@@ -49,7 +49,8 @@ func (da *pluginAPIAdapter) GetDirectChannel(userID1, userID2 string) (*model.Ch
 }
 
 func (da *pluginAPIAdapter) CreatePost(post *model.Post) error {
```

Comment:
It is ok because for now we only use this in this context, but this function name can be misleading, because you can try to create a post without wanting to get that processed. Probably would be more clear something like:

```go
func (da *pluginAPIAdapter) CreatePost(post *model.Post) error {
    return da.client.Post.CreatePost(post)
}

func (da *pluginAPIAdapter) CreatePostWithEmbededCard(post *model.Post) error {
    postEmbed, _ := postWithBoardsEmbed(post, da.client.Configuration.GetConfig().FeatureFlags.BoardsUnfurl)
    return da.CreatePost(postEmbed)
}
```

Also, extracting the link every time, even whenever we already have the link is sounds sub-optimal, and also adds certain dependency on the structure of the message. In other hand, if you optionally pass the link, you can skip that extra step (if you want) and also remove that requirement of including explicitly the link in the message.

Reply:
I liked that the interface used was a subset of the real plugin interface. Now it will be something else. Not a huge deal but we're creating a new API for the unlikely case that the notification system would need to create posts that don't unfurl.  

Also, unfurl was not implemented by embedding a link.  It embeds the components of a link (workspace, board, card).  I might as well just do the embed in server code and keep the plugin API as is.

Reply:
I like your refactor, but you included a ton of not related changes with dependencies.

Reply:
They are related.  I needed to update server v6 dependency since an unfurl struct is in model there.  Then a `go mod tidy` did the rest.

Reply:
... and that necessitated a `make generate`

Reply:
Ohh, you are right.